File: plugin/com.hello.uxml.tools.ui/src/com/hello/uxml/tools/framework/events/KeyboardEventArgs.java
package com.hello.uxml.tools.framework.events;

/**
* Defines keyboard event arguments.
*
* @author ferhat@
*/
public class KeyboardEventArgs extends EventArgs {

  // Keyboard EventType constants
  public static final int KEY_DOWN = 0;
  public static final int KEY_UP = 1;
  public static final int MOUSE_MOVE = 2;
  public static final int MOUSE_ENTER = 3;
  public static final int MOUSE_EXIT = 4;

  /** Keyboard event type */
  private int eventType;

  private int keyCode;
  private int charCode;

  /**
   * Constructor
   */
  public KeyboardEventArgs() {
  }

  /**
   * Constructor
   */
  public KeyboardEventArgs(int eventType, int keyCode, int charCode) {
    this.eventType = eventType;
    this.keyCode = keyCode;
    this.charCode = charCode;
  }

  /**
   * Initializes arguments so the instance can be reused.
   */
  public void reset() {
    handled = false;
  }

  /**
   * Sets or returns mouse event type.
   */
  public int getEventType() {
    return eventType;
  }

  public void setEventType(int eventType) {
    this.eventType = eventType;
  }

  /**
   * Sets or returns key code.
   */
  public int getKeyCode() {
    return keyCode;
  }

  public void setKeyCode(int code) {
    keyCode = code;
  }

  /**
   * Sets or returns character code.
   */
  public int getCharCode() {
    return charCode;
  }

  public void setCharCode(int code) {
    charCode = code;
  }
}

File: framework/lib/keyboard_event_args.dart
part of uxml;

/**
* Defines mouse event arguments.
*
* @author ferhat@ (Ferhat Buyukkokten)
*/
class KeyboardEventArgs extends EventArgs {

  // Mouse EventType constants
  static const int KEY_DOWN = 0;
  static const int KEY_UP = 1;

  static const int KEYCODE_ENTER = 13;
  static const int KEYCODE_SPACE = 32;
  static const int KEYCODE_DOWN = 40;
  static const int KEYCODE_UP = 38;
  static const int KEYCODE_LEFT = 37;
  static const int KEYCODE_RIGHT = 39;
  static const int KEYCODE_TAB = 9;

  static bool shiftKey = false;
  static bool ctrlKey = false;
  static bool altKey = false;
  static bool metaKey = false;

  /** Keyboard event type */
  int _eventType;

  /** key code */
  int keyCode;

  /** character code */
  int charCode;

  /**
   * Constructor
   */
  KeyboardEventArgs(int eventType,
      int key, int character, UxmlElement source) : super(source) {
    _eventType = eventType;
    keyCode = key;
    charCode = character;
  }

  /**
   * Sets or returns mouse event type.
   */
  int get eventType {
    return _eventType;
  }

  set eventType(int eventType) {
    _eventType = eventType;
  }
}
